File: frontend/src/app/services/client.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Client } from '../models/client.model';
import { ApiService } from './api.service';

@Injectable({
    providedIn: 'root'
})
export class ClientService {
    constructor(private api: ApiService) { }

    getAll(): Observable<Client[]> {
        return this.api.get<Client[]>('clients');
    }

    getById(id: number): Observable<Client> {
        return this.api.get<Client>(`clients/${id}`);
    }

    create(client: Client): Observable<Client> {
        return this.api.post<Client>('clients', client);
    }

    update(id: number, client: Client): Observable<Client> {
        return this.api.put<Client>(`clients/${id}`, client);
    }

    delete(id: number): Observable<any> {
        return this.api.delete(`clients/${id}`);
    }
}

File: src/main/java/org/anpfacturationbackend/service/ClientService.java
package org.anpfacturationbackend.service;

import org.anpfacturationbackend.dto.ClientDTO;
import java.util.List;

public interface ClientService {
    List<ClientDTO> getAll();

    ClientDTO getById(@org.springframework.lang.NonNull Long id);

    ClientDTO create(ClientDTO dto);

    ClientDTO update(@org.springframework.lang.NonNull Long id, ClientDTO dto);

    void delete(@org.springframework.lang.NonNull Long id);
}

